cloudbridge/cloud/providers/gce/resources.py

```diff
@@ -1548,11 +1684,163 @@ def create_volume(self, placement, size=None, volume_type=None, iops=None):
             'type': vol_type,
             'sourceSnapshot': self.id
         }
-        operation = (self._provider.gce_compute
+        operation = (self._provider
+                         .gce_compute
                          .disks()
-                         .insert(
-                             project=self._provider.project_name,
-                             zone=placement,
-                             body=disk_body).execute())
+                         .insert(project=self._provider.project_name,
+                                 zone=placement,
+                                 body=disk_body)
+                         .execute())
         return self._provider.block_store.volumes.get(
             operation.get('targetLink'))
+
+
+class GCSObject(BaseBucketObject):
+
+    def __init__(self, provider, bucket, obj):
+        super(GCSObject, self).__init__(provider)
+        self._bucket = bucket
+        self._obj = obj
+
+    @property
+    def id(self):
+        return self._obj['id']
+
+    @property
+    def name(self):
+        return self._obj['name']
+
+    @property
+    def size(self):
+        return self._obj['size']
+
+    @property
+    def last_modified(self):
+        return self._obj['updated']
+
+    def iter_content(self):
+        # TODO: It's not clear what does this method do. Does it return an
+        # iterator for metadata fields?
+        raise NotImplementedError('Not Implemented')
```

Comment:
@chiniforooshan This allows the actual content of the object to be iterated through as a byte stream, for example as shown here: https://github.com/gvlproject/cloudbridge/blob/6da5b0057385d7e4ba23245a5db9d8139513663b/cloudbridge/cloud/base/resources.py#L604